File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countPartitions = function (nums) {
    let ans = 0
    let sum = 0
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i]
    }
    let temp = 0
    for (let i = 0; i < nums.length - 1; i++) {
        temp += nums[i]
        let diff = Math.abs(temp - Math.abs(sum - temp))
        if (diff % 2 == 0) {
            ans++
        }
    }
    return ans
};

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total = sum(nums)
        left = 0
        count = 0
        for i in range(len(nums) - 1):
            left += nums[i]
            right = total - left
            if (left + right) % 2 == 0:
                count += 1

        return count

